cli/cmd/publicdomain/ls.go

```diff
@@ -0,0 +1,86 @@
+package publicdomain
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rio/cli/pkg/clicontext"
+	"github.com/rancher/rio/cli/pkg/clientcfg"
+	"github.com/rancher/rio/cli/pkg/table"
+	"github.com/rancher/rio/pkg/namespace"
+	"github.com/rancher/rio/types/client/space/v1beta1"
+)
+
+type Ls struct {
+}
+
+func (l *Ls) Run(ctx *clicontext.CLIContext) error {
+	cluster, err := ctx.Cluster()
+	if err != nil {
+		return err
+	}
+	spaceClient, err := cluster.Client()
+	if err != nil {
+		return err
+	}
+	publicDomains, err := spaceClient.PublicDomain.List(&types.ListOpts{})
+	if err != nil {
+		return err
+	}
+	writer := table.NewWriter([][]string{
+		{"NAME", "{{.Name}}"},
+		{"DOMAIN", "{{.DomainName}}"},
+		{"TARGET", "{{. | formatTarget}}"},
+	}, ctx)
+	defer writer.Close()
+
+	w := wrapper{
+		clusterClient: cluster,
+		spaceClient:   spaceClient.Space,
+	}
+	writer.AddFormatFunc("formatTarget", w.FormatTarget)
+	for _, publicDomain := range publicDomains.Data {
+		publicDomain.DomainName = "https://" + publicDomain.DomainName
+		writer.Write(&publicDomain)
+	}
+	return writer.Err()
+}
+
+type wrapper struct {
+	clusterClient *clientcfg.Cluster
+	spaceClient   client.SpaceOperations
+}
+
+func (w wrapper) FormatTarget(obj interface{}) (string, error) {
+	if v, ok := obj.(*client.PublicDomain); ok {
+		w.clusterClient.DefaultWorkspaceName = v.TargetWorkspaceName
+		workspace, err := w.clusterClient.Workspace()
+		if err != nil {
+			return "", nil
+		}
+		workspaceClient, err := workspace.Client()
+		if err != nil {
+			return "", nil
+		}
+		spaces, err := w.spaceClient.List(&types.ListOpts{})
```

Comment:
This will do N calls.  So if I have 5 domains this will call List 5 times.  you can instead list all the stacks once up front and then after just search through the list.